Added Security feature with PIN verification and JUnit tests

Harmanpreet-Deo opened the pull request:
Added Security feature with PIN verification and account lockout.
All JUnit tests passed.

MissPinkElf1111 reviewed (changes requested):
need to add a method to return "attemptsleft" value, with test.

Harmanpreet-Deo commented:
@MissPinkElf1111 I have added the 'getAttemptsLeft' method and updated the test case. I have also added additional dependencies required for mock JUnit testing. Let me know if any other changes are needed!
